src/AutoRest.CSharp/Common/AutoRest/Plugins/CSharpGen.cs

```diff
@@ -32,14 +32,17 @@ public async Task<GeneratedCodeWorkspace> ExecuteAsync(CodeModel codeModel)
             }
             else if (Configuration.AzureArm)
             {
-                if (Configuration.MgmtTestConfiguration is not null)
+                if (Configuration.MgmtConfiguration.MgmtDebug.SkipCodeGen)
```

Comment:
I think `SkipSampleGen` is much useful than `SkipCodeGen` because sample relays on the latest sdk, If the SDK code changes, the generated samples might fail to build.